amethyst_core/src/hidden.rs

```diff
@@ -14,9 +14,32 @@ impl Component for Hidden {
 
 /// Like [Hidden](struct.Hidden.html), but can propagate through children when the [HideHierarchySystem](struct.HideHierarchySystem.html)
 /// is enabled in the [RenderBundle](struct.RenderBundle.html).
-#[derive(Clone, Debug, Default)]
-pub struct HiddenPropagate;
+#[derive(Clone, Debug)]
+pub struct HiddenPropagate {
+    pub(crate) was_propagated: bool,
+}
 
 impl Component for HiddenPropagate {
-    type Storage = FlaggedStorage<Self, NullStorage<Self>>;
+    type Storage = FlaggedStorage<Self>;
+}
+
+impl HiddenPropagate {
+    /// Creates an instance of HiddenPropagate.
+    pub fn new() -> Self {
+        Self {
+            was_propagated: false,
```

Comment:
kind of feel `is_propagated` is more natural to search for, and think the past tense in "propagated" is enough to capture the meaning

Reply:
It now sounds like present in passive voice, but maybe that's alright.. We should add a comment for that field then